packages/user/Config/ui/src/pages/packages.tsx

```diff
@@ -0,0 +1,180 @@
+                                                                                </Button>
+                                                                            </div>
+                                                                        );
+                                                                    }}
+                                                                </form.AppField>
+                                                            ),
+                                                        )}
+                                                    </div>
+                                                );
+                                            }}
+                                        </form.AppField>
+
+                                        <div className="flex justify-between py-2">
+                                            <Button
+                                                variant="outline"
+                                                onClick={(e) => {
+                                                    e.preventDefault();
+                                                    form.setFieldValue(
+                                                        "accounts",
+                                                        (accounts) => [
+                                                            ...accounts,
+                                                            "",
+                                                        ],
+                                                    );
+                                                }}
+                                            >
+                                                New source
+                                            </Button>
+                                            <div className="flex gap-2">
+                                                <form.AppForm>
+                                                    <form.SubmitButton
+                                                        labels={[
+                                                            "Save",
+                                                            "Saving",
+                                                        ]}
+                                                    />
+                                                </form.AppForm>
+                                            </div>
+                                        </div>
+                                    </form>
+                                </DialogHeader>
+                            </DialogContent>
+                        </Dialog>
+                    </div>
+                </div>
+                <div className="flex flex-col gap-2">
+                    {data.map((pack) => (
+                        <PackageItem
+                            pack={pack}
+                            onClick={(id) => {
+                                installPackages([id]);
```

Comment:
Rather than immediately installing, I think this should mark the package as scheduled for installation, with a separate button to apply all changes, to allow multiple packages to be installed or updated at once. It's not a big problem for a single producer, but multiple staged transactions might conflict with each other.